### ISSUE TITLE
Decimal types with scale > 28 cannot be read

### ISSUE THREAD

**Issue by MaceWindu** [DarkWanderer/ClickHouse.Client#230]
Sample type `Decimal(38, 29)`
Exponent calculated using `decimal` type and fails with overflow here: https://github.com/DarkWanderer/ClickHouse.Client/blob/master/ClickHouse.Client/Types/DecimalType.cs#L26

```cs
   at System.Number.ThrowOverflowException(TypeCode type)
   at System.Decimal.DecCalc.ScaleResult(Buf24* bufRes, UInt32 hiRes, Int32 scale)
   at System.Decimal.DecCalc.VarDecMul(DecCalc& d1, DecCalc& d2)
   at System.Decimal.op_Multiply(Decimal d1, Decimal d2)
   at ClickHouse.Client.Utility.MathUtils.ToPower(Decimal value, Int32 power) in D:\a\ClickHouse.Client\ClickHouse.Client\ClickHouse.Client\Utility\MathUtils.cs:line 40

```

```cs
ClickHouse.Client.dll!ClickHouse.Client.Utility.MathUtils.ToPower(decimal value, int power) Line 40
ClickHouse.Client.dll!ClickHouse.Client.Types.DecimalType.Scale.set(int value) Line 26
ClickHouse.Client.dll!ClickHouse.Client.Types.DecimalType.Parse(ClickHouse.Client.Types.Grammar.SyntaxTreeNode node, System.Func<ClickHouse.Client.Types.Grammar.SyntaxTreeNode, ClickHouse.Client.Types.ClickHouseType> parseClickHouseTypeFunc, ClickHouse.Client.TypeSettings settings) Line 62
ClickHouse.Client.dll!ClickHouse.Client.Types.TypeConverter.ParseClickHouseType(ClickHouse.Client.Types.Grammar.SyntaxTreeNode node, ClickHouse.Client.TypeSettings settings) Line 131
ClickHouse.Client.dll!ClickHouse.Client.Types.TypeConverter.ParseClickHouseType(string type, ClickHouse.Client.TypeSettings settings) Line 119
ClickHouse.Client.dll!ClickHouse.Client.ADO.Readers.ClickHouseDataReader.ReadHeaders() Line 215
ClickHouse.Client.dll!ClickHouse.Client.ADO.Readers.ClickHouseDataReader.ClickHouseDataReader(System.Net.Http.HttpResponseMessage httpResponse, ClickHouse.Client.TypeSettings settings) Line 35
ClickHouse.Client.dll!ClickHouse.Client.ADO.ClickHouseCommand.ExecuteDbDataReaderAsync(System.Data.CommandBehavior behavior, System.Threading.CancellationToken cancellationToken) Line 142

```

**Comment by DarkWanderer** [comment]
Thanks for report. Should be fixed in [6.2.0](https://github.com/DarkWanderer/ClickHouse.Client/releases/tag/6.2.0)

**Comment by MaceWindu** [comment]
Fixed
